### DISCUSSION TITLE
Is there a way to add objectFit='cover' like behaviour to the new next/image component?

### DISCUSSION THREAD

**Discussion by la55u** [vercel/next.js#18385] [Ideas]
# Feature request

## Is your feature request related to a problem? Please describe.
I was trying to replace the `Image` component from `chakra-ui` with the new `next/image` component. The chakra-ui (and I'm sure many other libraries behave similarly) component looks like this: 
``` 
import {Image} from "@chakra-ui/core";

<Image
  w="full"
  h="auto"
  maxH="300px"
  objectFit="cover"
  src="/img/sample.jpg"
  alt="cover image"
 loading="lazy"
 /> 
```

The above behaviour is currently not possible due to the lack of max-height and object-fit: cover support for `next/image`
```
import NextImage from "next/image";

<NextImage
  width={600}
  height={300}
  src="/img/sample.jpg"
  alt="cover image"
/>
```
I assume a pretty common use-case for images is to show them with (one or more) fixed size but since they come from a CMS the exact size is unknown. To overcome this issue I usually set `max-width` or `max-height` on the image and put `object-fit: cover` on it as well.  

## Describe the solution you'd like
I propose adding support for maxWidth and maxHeight props as well as objectFit.

## Describe alternatives you've considered

I tried adding these CSS properties via the style prop but they are not applied on the generated element, neither with the `unsized` prop nor without it.
```
  style={{ objectFit: "cover", maxHeight: 300 }}
```

**Comment by smontlouis** [comment]
Did you find a solution ? @la55u

**Comment by evantahler** [comment]
It seems you can't even pass a `ref` to the image component to apply the styles later with a `useEffect` hook...

**Comment by Timer** [comment]
Yes! On `next@canary` you can pass `objectFit="cover"` as a prop.

**Reply by evantahler** [reply]
Excellent!

**Reply by gurupal** [reply]
Amazing

**Reply by louisbarclay** [reply]
Love it

**Reply by SteplerPaulo** [reply]
Amazing. It should be on the document, BTW

**Reply by AngeloCore** [reply]
amazing

**Reply by DevUday2705** [reply]
Thanks bro :)

**Comment by IHATECOFFEE1** [comment]
You can use obejectFit = cover property as long as u have the layout = fill.  [https://nextjs.org/docs/basic-features/image-optimization](url)

**Reply by Al-Faiz-Ahmed** [reply]
Layout fill has been deprected by next js 13  :(

**Reply by mrahmadhassankhan** [reply]
Instead of writing layout : "fill" only write fill in next js 15.1.7 this will work for you

**Comment by MoncefDeveloper** [comment]
next/image changed in version 13,
" Removes objectFit prop in favor of style or className "

**Comment by mrahmadhassankhan** [comment]
if you are using tailwind then simply done like this in Next js 15.1.7

`   <Image
          src="/example.jpg"
          alt="internship"
          fill
          priority
          quality={100}
          className="object-cover"
        />`
